job_templates/ci_job.xml.em

```diff
@@ -132,27 +132,17 @@ fi
 if [ "$CI_USE_WHITESPACE_IN_PATHS" = "true" ]; then
   export CI_ARGS="$CI_ARGS --white-space-in sourcespace buildspace installspace workspace"
 fi
-export CI_ARGS="$CI_ARGS --ignore-rmw rmw_connext_dynamic_cpp"
-# TODO(asorbini) `rmw_connext_cpp` is still the default for foxy.
-if [ -n "$CI_ROS_DISTRO" -a \( "$CI_ROS_DISTRO" = foxy \) ]; then
-  if [ "$CI_USE_CONNEXTDDS" = "false" ]; then
-    export CI_ARGS="$CI_ARGS rmw_connext_cpp"
-  fi
-else
-  # Always ignore `rmw_connext_cpp` in favor of `rmw_connextdds` for newer releases.
-  export CI_ARGS="$CI_ARGS rmw_connext_cpp"
-  if [ "$CI_USE_CONNEXTDDS" = "false" ]; then
-    export CI_ARGS="$CI_ARGS rmw_connextdds"
-  fi
+if [ "$CI_USE_CONNEXTDDS" = "false" ]; then
+  export CI_ARGS="$CI_ARGS --ignore-rmw rmw_connextdds"
 fi
 if [ "$CI_USE_CYCLONEDDS" = "false" ]; then
-  export CI_ARGS="$CI_ARGS rmw_cyclonedds_cpp"
+  export CI_ARGS="$CI_ARGS --ignore-rmw rmw_cyclonedds_cpp"
 fi
 if [ "$CI_USE_FASTRTPS_STATIC" = "false" ]; then
-  export CI_ARGS="$CI_ARGS rmw_fastrtps_cpp"
+  export CI_ARGS="$CI_ARGS --ignore-rmw rmw_fastrtps_cpp"
 fi
 if [ "$CI_USE_FASTRTPS_DYNAMIC" = "false" ]; then
-  export CI_ARGS="$CI_ARGS rmw_fastrtps_dynamic_cpp"
+  export CI_ARGS="$CI_ARGS --ignore-rmw rmw_fastrtps_dynamic_cpp"
```

Comment:
These changes aren't connext related. This does look like a bug fix, since these rmw would only be properly ignored whhen connext is also ignored to provide the initial `--ignore-rmw` argument but I think that this change should be split out into a separate PR.